File: HaberUygulamaJava/app/src/main/java/com/example/haberuygulamajava/servis/HaberService.java
package com.example.haberuygulamajava.servis;

import com.example.haberuygulamajava.model.HaberModel;
import com.example.haberuygulamajava.model.HaberTuruModel;
import com.example.haberuygulamajava.model.YorumInsertRequest;
import com.example.haberuygulamajava.model.YorumModel;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Body;
import retrofit2.http.Query;

public interface HaberService {

    @GET("haber_haberler-get.php")
    Call<List<HaberModel>> getHaberler();

    @GET("haber_yorumlar-get.php")
    Call<List<YorumModel>> getYorumlar(@Query("haber_id") int haberId);

    @POST("haber_yorumlar-insert.php")
    Call<ApiResponse> insertYorum(@Body YorumInsertRequest request);

    @GET("haber_haberturleri-get.php")
    Call<List<HaberTuruModel>> getKategoriler();

    @GET("haber_haberler-son3-get.php")
    Call<List<HaberModel>> getSon3Haber();

    @GET("haber_haberler-sondakika-get.php")
    Call<List<HaberModel>> getSonDakikaHaberler();

    @GET("haber_haberler-gundem-get.php")
    Call<List<HaberModel>> getGundemHaberler();
}

File: HaberUygulamaKotlin/app/src/main/java/com/example/haberuygulama/servis/HaberService.kt
package com.example.haberuygulama.servis

import com.example.haberuygulama.model.HaberModel
import com.example.haberuygulama.model.HaberTuruModel
import com.example.haberuygulama.model.YorumInsertRequest
import com.example.haberuygulama.model.YorumModel
import retrofit2.Call
import retrofit2.http.*


interface HaberService {

    @GET("haber_haberler-get.php")
    fun getHaberler(): Call<List<HaberModel>>


    @GET("haber_yorumlar-get.php")
    fun getYorumlar(@Query("haber_id") haberId: Int): Call<List<YorumModel>>

    @POST("haber_yorumlar-insert.php")
    fun insertYorum(@Body request: YorumInsertRequest): Call<ApiResponse>

    @GET("haber_haberturleri-get.php")
    fun getKategoriler(): Call<List<HaberTuruModel>>

    @GET("haber_haberler-son3-get.php")
    fun getSon3Haber(): Call<List<HaberModel>>

    @GET("haber_haberler-sondakika-get.php")
    fun getSonDakikaHaberler(): Call<List<HaberModel>>

    @GET("haber_haberler-gundem-get.php")
    fun getGundemHaberler(): Call<List<HaberModel>>


}
